dev/src/write-batch.ts

```diff
@@ -517,12 +560,16 @@ export class WriteBatch {
    * @param commitOptions.transactionId The transaction ID of this commit.
    * @param commitOptions.requestTag A unique client-assigned identifier for
    * this request.
+   * @param useBatch Whether to use BatchWrite API instead of CommitRequest.
    * @returns  A Promise that resolves when this batch completes.
    */
-  async commit_(commitOptions?: {
-    transactionId?: Uint8Array;
-    requestTag?: string;
-  }): Promise<WriteResult[]> {
+  async commit_(
+    commitOptions?: {
+      transactionId?: Uint8Array;
+      requestTag?: string;
+    },
+    useBatch = false
+  ): Promise<WriteResult[] | BatchWriteResult[]> {
```

Comment:
I think this can always return `BatchWriteResult`. `commit()` can drop the unused Status.

Reply:
N/A since I separated out the logic